client/src/offline-settings/mdn-worker.tsx

```diff
@@ -273,6 +274,19 @@ export class MDNWorker {
           this.updating = false;
         }
 
+        if (this.updating && !this.keepAlive) {
```

Comment:
Can we clean up this entire function (setStatus) ? It's quite hard to read now with all these new flags and conditions